docs: improve cd explanation (#129)
canonical/starflow: docs/explanation/continuous-deployment-documentation.rst

1. A contributor proposes a document.
2. Both technical authors review the document.
3. The continuous integration (CI) system runs the documentation test suite against the
   document. If product code was changed, the CI system runs the full test suite.
4. A technical author approves and merges the document.

> 5. Read the Docs publishes the update to ``https://<docs>/latest``

Reviewer: Number with `#.`?

Author: 08e22e0

Current releases
~~~~~~~~~~~~~~~~

.. image:: assets/continuous-deployment-release.svg
    :loading: lazy
    :alt: A Git tree history of the release branch, in parallel with the history of the main branch. Each commit to the hotfix branch maps to a new publication for the hotfix's documentation version on Read the Docs. Documentation commits are represented in both histories, and publish separately until recombined in main.

All documentation changes in a current release's branch are immediately deployed to the

> corresponding documentation version on Read the Docs. The documentation process is

Reviewer: ```suggestion
corresponding documentation version on Read the Docs. The process is
```
Implied from the previous sentence.

Author: 7ce5099

#. On Read the Docs, the TA creates documentation version 1.3, sourced from the
   ``hotfix/1.3`` branch.
#. The TA redirects the prior minor documentation versions to documentation version 1.3.
#. Read the Docs publishes the new version for that release, at ``https://<docs>/1.3``

To make a documentation change for 1.3:

#. A contributor proposes a document.
#. Both technical authors review the document.

> #. The CI checks run the documentation checks against the document. If product code was

Reviewer: ```suggestion
#. The CI system runs the documentation checks against the document. If product code was
```
More consistent with the next sentence

Author: 7ce5099

   changed, the CI system runs the full test suite.
#. A technical author approves and merges the document.
#. Read the Docs publishes the updated docs to ``https://<docs>/1.3``

Key factors
~~~~~~~~~~~

**Versioned documentation is deployed from hotfix branches.** The most important trait

> in the process is that a current release's docs are republished with every commit. Every

Reviewer: ```suggestion
of the process is that a current release's docs are republished with every commit. Every
```
Nit

Author: 7ce5099

commit to the release's hotfix branch republishes the documentation. They *do not*
*wait* for a new release. That way, improvements to the documentation are continuous.

If documentation is deployed parallel with the software's releases, why not merge to

> ``main`` and cherry pick changes to hotfix branches, instead? First, because hotfixes

Reviewer: ```suggestion
If documentation is deployed in parallel with the software's releases, why not merge to
``main`` and cherry pick changes to hotfix branches instead? First, because hotfixes
```
More nits

Author: 7ce5099

Users
~~~~~

- When I go to ``https://<docs>/latest``, I see the latest documents from the ``main``
  branch.
- When I go to ``https://<docs>/1.3``, I see the latest documents for version 1.3.
- If there's a correction to the documents for version 1.3, I see it immediately in
  ``https://<docs>/1.3``.
- If there's a patch to version 1.3, the patch description is available at
  ``https://<docs>/1.3``.

Contributors
~~~~~~~~~~~~

- When I open a PR that only changes documentation, the CI system only runs the
  documentation checks.

> - When reviewing my PR, I can expect that simple changes are merged within a day after

Reviewer: ```suggestion
- When reviewing my PR, I can expect that simple changes are merged within a day of receiving
```

Author: 7ce5099
